compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7179,10 +7179,18 @@ private STNode parseCheckExpression(boolean isRhsExpr, boolean allowActions, boo
         STNode checkingKeyword = parseCheckingKeyword();
```

Comment:
Now that we have an optional `on-fail-check` at the end of `check-expr`, we need to add a new path as the next rule in the `seekMatchInExpressionRhs()`  for `check-expr`. In order to distinguish it at that method we need to a context to identify whether we are in the `check-expr`. 
```suggestion
        startContext(ParserRuleContext.CHECK_EXPRESSION);
        STNode checkingKeyword = parseCheckingKeyword();
```

Reply:
1. Once we have the context, we can add the following in `seekMatchInExpressionRhs()` to point the next rule as `ParserRuleContext.ON_FAIL_CHECK`.
```java
        } else if (parentCtx == ParserRuleContext.CHECK_EXPRESSION) {
            STToken nextToken = this.tokenReader.peek(lookahead);
            if (nextToken.kind == SyntaxKind.ON_KEYWORD) {
                nextContext = ParserRuleContext.ON_FAIL_CHECK;
            } else {
                nextContext = ParserRuleContext.CHECK_EXPR_END;
            }
```

Reply:
2. Need to define the next rule for `ParserRuleContext.ON_FAIL_CHECK` as `ParserRuleContext.ON_KEYWORD`.
3. The next rule for `ParserRuleContext.CHECK_EXPR_END` will be,
```java
endContext();
ParserRuleContext.EXPRESSION_RHS;
```

Reply:
4. We need to continue the path created under 2 in the following manner. ON_KEYWORD -> FAIL_KEWORD -> VARIABLE_NAME -> RIGHT_ARROW -> ERROR_CONSTRUCTOR

Reply:
Updated